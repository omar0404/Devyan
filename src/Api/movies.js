import axios from 'axios';
import Config from 'react-native-config'
const { API_KEY, API_BASE_URL } = Config
const getMovies = (page) => {
  const params = {
    api_key: API_KEY,
    page,
  };
  return axios.get(API_BASE_URL, { params });
};

export { getMovies };
