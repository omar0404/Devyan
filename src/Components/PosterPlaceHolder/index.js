import PosterPlaceHodler from './PosterPlaceHolder';
export default PosterPlaceHodler;
