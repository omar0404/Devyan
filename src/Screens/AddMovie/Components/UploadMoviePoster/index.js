import UploadMoviePoster from './UploadMoviePoster';
export default UploadMoviePoster;
