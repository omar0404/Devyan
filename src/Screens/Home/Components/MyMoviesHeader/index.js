import MyMoviesHeader from './MyMoviesHeader';
export default MyMoviesHeader;
