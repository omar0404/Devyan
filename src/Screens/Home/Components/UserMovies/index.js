import UserMovies from './UserMovies';
export default UserMovies;
