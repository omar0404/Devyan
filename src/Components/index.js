import Section from './Section';
import PosterPlaceHolder from './PosterPlaceHolder';
import MovieCard from './MovieCard';
export {MovieCard, Section, PosterPlaceHolder};
