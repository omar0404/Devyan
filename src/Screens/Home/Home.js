import React from 'react';
import {FlatList} from 'react-native';
import {MyMovies, AllMovies} from './Components';
import styles from './styles';
const Home = () => {
  return (
    <FlatList
      style={styles.container}
      ListHeaderComponent={<MyMovies />}
      ListFooterComponent={<AllMovies />}
    />
  );
};
export default Home;
