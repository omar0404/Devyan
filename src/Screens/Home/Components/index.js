import MyMovies from './MyMovies';
import AllMovies from './AllMovies';
import UserMovies from './UserMovies';
import MyMoviesHeader from './MyMoviesHeader';
export {MyMovies, AllMovies, UserMovies, MyMoviesHeader};
