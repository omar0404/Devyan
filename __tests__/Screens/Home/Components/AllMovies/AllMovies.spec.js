import React from 'react';

import {
  renderFooter,
  renderMovie,
  handleEndReached,
  handleKeyExtractor,
} from 'Devyan/src/Screens/Home/Components/AllMovies/AllMovies';
import {shallow} from 'enzyme';
import {AllMovies} from 'Devyan/src/Screens/Home/Components';
import {create, act} from 'react-test-renderer';
import {getMovies} from 'Devyan/src/Api/movies';
describe('AllMovies Footer render function', () => {
  it('should render loading indicator while fetching new movies', () => {
    expect(renderFooter(true)).toBeDefined();
  });
  it('should render nul when there is no fetching', () => {
    expect(renderFooter(false)).toBeNull();
  });
});

describe('Movies list reached to the end', () => {
  it('should increment page number', () => {
    const prevPage = 1;
    const setPage = jest.fn((cb) => {
      cb(prevPage);
    });
    handleEndReached(setPage)();
    expect(setPage).toBeCalled();
  });
});
describe('rendering movie', () => {
  it('should render movie card correctly', () => {
    const props = {
      item: {},
    };
    const onLikePress = jest.fn();
    const onMoviePress = jest.fn();
    expect(
      shallow(renderMovie(onLikePress, {}, onMoviePress)(props)),
    ).toMatchSnapshot();
  });
});

describe('Movies List Key Extractor', () => {
  it('should return unique key for movie  component', () => {
    const movies = [{id: 345}, {id: 34512}, {id: 29929}];
    const expectedMoviesUniqueKeys = [345, 34512, 29929];
    const result = movies.map((movie) => handleKeyExtractor(movie));
    expect(expectedMoviesUniqueKeys).toEqual(result);
  });
});

describe('AllMovies Component', () => {
  it('should render AllMovies Component correctly', async () => {
    const wrapper = create(<AllMovies />);
    await act(() => getMovies());
    expect(wrapper).toMatchSnapshot();
  });
});
