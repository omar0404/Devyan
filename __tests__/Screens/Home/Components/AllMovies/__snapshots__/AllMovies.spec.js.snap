// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllMovies Component should render AllMovies Component correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "auto",
        },
        Object {
          "marginTop": 50,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#991F25",
          "height": 15,
          "width": 5,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
          "fontSize": 18,
          "marginLeft": 8,
        }
      }
    >
      All Movies
    </Text>
  </View>,
  <RCTScrollView
    ListFooterComponent={null}
    data={
      Array [
        Object {
          "name": "Batman",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "height": 200,
                "marginRight": 20,
                "width": 200,
              },
              Object {
                "alignSelf": "center",
                "height": 400,
                "marginVertical": 20,
                "width": "90%",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "http://image.tmdb.org/t/p/w500undefined",
              }
            }
            style={
              Object {
                "flex": 2,
                "height": null,
                "width": null,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "marginTop": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "white",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgb(255, 255, 255)",
                    "flex": 1,
                  }
                }
              />
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "#991F25",
                    "flex": 1,
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "textAlign": "right",
                  }
                }
              />
            </View>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "gray",
                  "marginTop": 12,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;

exports[`rendering movie should render movie card correctly 1`] = `ShallowWrapper {}`;
